File: mac/lib/pickmath.js
// How far the 🎲 picker spins so it always stops exactly on the chosen name (same as Windows).
function steps(count, start, winner) {
  const laps = count <= 3 ? 8 : count < 10 ? 3 : 1;
  const base = Math.max(22, laps * count);
  return base + (((winner - (start + base) % count) + count) % count);
}
const landing = (count, start, s) => (start + s) % count;
if (typeof module !== 'undefined') module.exports = { steps, landing };

File: windows/PickMath.cs
namespace Remco;

/// <summary>How far the 🎲 picker spins so it always stops exactly on the chosen name.</summary>
internal static class PickMath
{
    public static int Steps(int count, int start, int winner)
    {
        int laps = count <= 3 ? 8 : count < 10 ? 3 : 1;
        int baseSteps = Math.Max(22, laps * count);
        return baseSteps + ((winner - (start + baseSteps) % count) + count) % count;
    }

    /// <summary>The name index shown when the spin ends.</summary>
    public static int Landing(int count, int start, int steps) => (start + steps) % count;
}
